akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -233,6 +252,24 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
     rec(if (path.startsWithSlash) path.tail else path)
   }
 
+  /**
+   * Check against directory traversal attempts by making sure that the final is a "true child"
+   * of the given base directory.
+   *
+   * Returns "" if the finalPath is suspicious and the canonical path otherwise.
+   */
+  private def checkIsSafeDescendant(basePath: String, finalPath: String, log: LoggingAdapter): String = {
+    val baseFile = new File(basePath)
+    val finalFile = new File(finalPath)
+    val canonicalFinalPath = finalFile.getCanonicalPath
+
+    if (!canonicalFinalPath.startsWith(baseFile.getCanonicalPath)) {
+      log.warning(s"[$finalFile] points to a location that is not part of [$baseFile]. This might be a directory " +
+        "traversal attempt.")
+      ""
```

Comment:
Otherwise, there might be a higher risk of revealing some information if the response is not the same as if the file is just missing.